modules/demo.xql

```diff
@@ -28,7 +28,7 @@ declare function demo:run-tests($node as node(), $model as map(*)) {
 };
 
 declare function demo:display-source($node as node(), $model as map(*), $lang as xs:string?, $type as xs:string?) {
-    let $source := replace($node/string(), "^\s*(.*)\s*$", "$1")
+    let $source := replace(replace($node/string(),'\s+$',''),'^\s+','')
```

Comment:
That looks like a trim equivalent expression
How about `normalize-space`? Shouldn't that also work?

Reply:
It might be enough to write
```xquery
let $source := normalize-space($node)
```

Reply:
```xquery
let $n := <a>
    d
    </a>

return [
    normalize-space($n),
    replace(replace($n/string(),'\s+$',''),'^\s+','')
]
```

Reply:
No, normalize removes double white space everywhere in the text;

If you do that, then the formatted queries will be un-formatted showed in the HTML;

We want to preserve the query internal \n 's :-) 
Only the leading/post whitespace should go.

Somehow.... path does not have a 'strip' function equivalent to what is available in java.

Reply:
I see

Reply:
`replace($node/string(), "^\s*(.+)\s*$", "$1")` this should do the trick then

Reply:
because that regex does _not_ match the empty string